libgpos/src/memory/CMemoryPoolManager.cpp

```diff
@@ -93,12 +94,45 @@ CMemoryPoolManager::CMemoryPoolManager
 //
 //---------------------------------------------------------------------------
 GPOS_RESULT
-CMemoryPoolManager::EresInit()
+CMemoryPoolManager::EresInit
```

Comment:
Seems like we need to break the dependency of the `CMemoryPoolManager` with `CMemoryPoolMalloc`. 

How to do `alloc` or `free` should be handled by the `CMemoryPoolMalloc` or `CMemoryPoolCustomAlloc`, and shouldn't be `CMemoryPoolManager`'s reponsibility to decide based on NULLability of the parameters.

I am wondering whether we should expose the `CMemoryPoolManager::ErasInit()` to take a `CMemoryPool` as parameter, instead of let the `CMemoryPoolManager` decide what kind of `CMemoryPool` to initialize?

I tried to avoid a class having an enum, then decide internally which implementation to call based on that enum. I would rather clearly separate the responsibilities, and only let `CMemoryPoolManager` manage a given type of `CMemoryPool` instead of knowing the details of how that pool allocate/free the memory.

Make sense?